lib/rubygems/commands/generate_index_command.rb

```diff
@@ -1,86 +1,51 @@
 # frozen_string_literal: true
 
 require_relative "../command"
-require_relative "../indexer"
 
-##
-# Generates a index files for use as a gem server.
-#
-# See `gem help generate_index`
-
-class Gem::Commands::GenerateIndexCommand < Gem::Command
-  def initialize
-    super "generate_index",
-          "Generates the index files for a gem server directory",
-          :directory => ".", :build_modern => true
+unless defined? Gem::Commands::GenerateIndexCommand
+  class Gem::Commands::GenerateIndexCommand < Gem::Command
+    module RubygemsTrampoline
+      def description # :nodoc:
+        <<~EOF
+          The server command has been moved to the rubygems-server gem.
+        EOF
+      end
 
-    add_option "-d", "--directory=DIRNAME",
-               "repository base dir containing gems subdir" do |dir, options|
-      options[:directory] = File.expand_path dir
-    end
+      def execute
+        alert_error "Install the rubygems-server gem for the server command"
+      end
 
-    add_option "--[no-]modern",
-               "Generate indexes for RubyGems",
-               "(always true)" do |value, options|
-      options[:build_modern] = value
+      def invoke_with_build_args(args, build_args)
+        name = "rubygems-generate_index"
+        spec = begin
+          Gem::Specification.find_by_name(name)
+        rescue Gem::LoadError
+          require "rubygems/dependency_installer"
+          Gem.install(name, Gem::Requirement.default, Gem::DependencyInstaller::DEFAULT_OPTIONS).find {|s| s.name == name }
```

Comment:
Any idea which gem source is used in here?

Reply:
I believe it is the default ?